### PR TITLE
Small optimizations in msgpack loading

### PR DESCRIPTION
**Pull request by AlexBrownAMD**
Reduces data copying and comparisons when reading arrays, maps, and enums.

### CODE REVIEW

File: Tensile/Source/lib/include/Tensile/Serialization/Predicates.hpp
Comment (nakajee) [inline Tensile/Source/lib/include/Tensile/Serialization/Predicates.hpp:227]: Can we remove the above enumeration?

Comment (nakajee) [reply Tensile/Source/lib/include/Tensile/Serialization/Predicates.hpp:227]: Please fix static analysis errors.

Comment (AlexBrownAMD) [reply Tensile/Source/lib/include/Tensile/Serialization/Predicates.hpp]: Fixed formatting. The other enumeration is likely still used by the YAML codepath, I haven't changed that part of code since it

Comment (nakajee) [reply Tensile/Source/lib/include/Tensile/Serialization/Predicates.hpp]: > Fixed formatting. The other enumeration is likely still used by the YAML codepath, I haven't changed that part of code.

Thanks for your update. We can keep the enumeration (I felt it was redundant...).
Why some tests are failing?
Is it caused by your change?

**Review by babakpst — CHANGES_REQUESTED** [review]
Blocked until FC merges are complete.

File: Tensile/Source/lib/include/Tensile/Serialization/Predicates.hpp
Comment (cgmb) [inline Tensile/Source/lib/include/Tensile/Serialization/Predicates.hpp:243]: If `Predicates.hpp` is included in N translation units, then N copies of this hash table will have to be initialized upon loading the Tensile library.

Comment (cgmb) [reply Tensile/Source/lib/include/Tensile/Serialization/Predicates.hpp:243]: We did this sort of thing with a regex in the rocsolver benchmark suite and it was a surprising source of slowdown. Once it was indirectly included in a few dozen translation units, it reached something like 10% of the runtime of the program when we were testing small matrix sizes.

**Comment by babakpst** [comment]
Closing this PR, as this is not a priority now.

**Pull request closed without merging**
